lab-3-functional-testing/src/test/kotlin/io/github/e1turin/fandom/common/FandomAuthPageTest.kt

```diff
@@ -0,0 +1,78 @@
+package io.github.e1turin.fandom.common
+
+import io.github.e1turin.fandom.getFandomProfileLogin
+import io.github.e1turin.fandom.getFandomProfilePassword
+import io.github.e1turin.fandom.pages.*
+import io.kotest.core.spec.style.stringSpec
+import io.kotest.matchers.shouldBe
+import org.openqa.selenium.ElementNotInteractableException
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.FluentWait
+import java.time.Duration
+
+
+fun buildAuthPageTests(driverSetup: WebDriver) = stringSpec {
+    lateinit var driver: WebDriver
+    lateinit var someFandomPage: FandomPage
+
+    val randomSiteURL = "https://matrix.fandom.com/wiki/Neo"
+
+    beforeTest {
+        driver = driverSetup
+        someFandomPage = object : FandomPage(driver, randomSiteURL) {
+            override fun status(): PageStatus = onThisPageIf(title().contains("Neo"))
+        }
+    }
+
+    afterTest {
+        driver.quit()
+    }
+
+    fun onAuthPage(steps: FandomAuthPage.() -> Unit) {
```

Comment:
Because I need somehow instantiate the page (generic parameter) in function. I think I could use reified type, but it's complicate for understanding the essence of tests on the first look. Also, the problem is in passing somehow `driver` parameter to page instance, so I can't see an easy way to generalize such util functions.